src/core/meshcop/border_agent.hpp

```diff
@@ -302,13 +302,22 @@ class BorderAgent : public InstanceLocator, private NonCopyable
     void HandleTimeout(void);
 
 #if OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE
-    void        RestartAfterRemovingEphemeralKey(void);
     void        HandleEphemeralKeyTimeout(void);
     void        InvokeEphemeralKeyCallback(void);
     static void HandleSecureAgentStopped(void *aContext);
     void        HandleSecureAgentStopped(void);
 #endif
 
+    enum RestartReason : uint8_t
+    {
+        kRestartOnPskcChanged,  ///< Restart to apply new pskc.
+        kRestartOnDisconnected, ///< Restart after disconnected, apply new pskc if it's changed during active session
+        kRestartOnEphemeralKeyRemoved, ///< Restart to apply pskc when ephemeral key is removed
```

Comment:
Add guard for the `enum` value (so we can use `switch()` will not get warning about unhandled `case`?
```suggestion
#if OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE
        kRestartOnEphemeralKeyRemoved,
#endif 
```